include/SatEncoder.hpp

```diff
@@ -55,37 +73,18 @@ class SatEncoder {
         void from_json(const json& j);
     };
 
-    class QState {
+    struct QState {
         unsigned long                  n;
         std::vector<std::vector<bool>> x;
         std::vector<std::vector<bool>> z;
         std::vector<int>               r;
         std::size_t                    prevGenId;
 
-    public:
-        void SetN(unsigned long n);
-
-        [[nodiscard]] const std::vector<std::vector<bool>>& GetX() const;
-
-        void SetX(const std::vector<std::vector<bool>>& x);
-
-        void SetZ(const std::vector<std::vector<bool>>& z);
-
-        void SetR(const std::vector<int>& r);
-
-        [[nodiscard]] const std::size_t& GetPrevGenId() const;
-
-        void SetPrevGenId(const std::size_t& prev_gen_id);
-
         [[nodiscard]] std::vector<std::vector<bool>> getLevelGenerator() const;
-
-        void applyCNOT(unsigned long control, unsigned long target);
-
-        void applyH(unsigned long target);
-
-        void applyS(unsigned long target);
-
-        void printStateTableau();
+        void                                         applyCNOT(unsigned long control, unsigned long target);
+        void                                         applyH(unsigned long target);
+        void                                         applyS(unsigned long target);
+        void                                         printStateTableau();
```

Comment:
Code format should be fixed by running clang-format as indicated by the corresponding CI check.